redaxo/src/core/lib/form/form.php

```diff
@@ -253,35 +253,49 @@ protected function getValue($name)
     /**
      * Callbackfunktion, damit in subklassen der Value noch beeinflusst werden kann
      * kurz vorm speichern.
+     *
+     * @param string          $fieldsetName
+     * @param string          $fieldName
+     * @param string|int|null $fieldValue
+     *
+     * @return string|int|null
      */
     protected function preSave($fieldsetName, $fieldName, $fieldValue, rex_sql $saveSql)
     {
         static $setOnce = false;
 
         if (!$setOnce) {
-            $fieldnames = $this->sql->getFieldnames();
+            $this->setGlobalSqlFields($saveSql);
+            $setOnce = true;
+        }
 
-            if (in_array('updateuser', $fieldnames)) {
-                $saveSql->setValue('updateuser', rex::getUser()->getValue('login'));
-            }
+        return $fieldValue;
+    }
 
-            if (in_array('updatedate', $fieldnames)) {
-                $saveSql->setDateTimeValue('updatedate', time());
-            }
+    /**
+     * Sets the sql fields `updateuser`, `updatedate`, `createuser` and `createdate` (if available).
+     */
+    private function setGlobalSqlFields(rex_sql $saveSql): void
+    {
+        $fieldnames = $this->sql->getFieldnames();
 
-            if (!$this->isEditMode()) {
-                if (in_array('createuser', $fieldnames)) {
-                    $saveSql->setValue('createuser', rex::getUser()->getValue('login'));
-                }
+        if (in_array('updateuser', $fieldnames)) {
```

Comment:
strict-in-array?

Reply:
theoretisch ja, wobei der code nur verschoben wurde und vorher schon so war